Source/Charts/Highlight/ChartHighlighter.swift

```diff
@@ -73,17 +73,11 @@ open class ChartHighlighter : NSObject, Highlighter
         
         guard let data = self.data else { return vals }
         
-        for i in 0 ..< data.dataSetCount
-        {
-            guard
-                let dataSet = data.getDataSetByIndex(i),
-                dataSet.isHighlightEnabled      // don't include datasets that cannot be highlighted
-                else { continue }
-            
-
+        for (i, set) in zip(data.indices, data) where set.isHighlightEnabled
```

Comment:
I like this advanced use. Does this come with the collection adoption?

Reply:
`for d in data` comes from `Sequence` adoption which `Collection` inherits. This also allows the zip(:) function to be used.